framework/include/materials/MaterialData.h

```diff
@@ -155,8 +158,13 @@ class MaterialData
   ///@}
 
   /**
-   * Resizes the properties to the specified size
-   * @param size The size of the properties to set
+   * Resizes the number of properties to the specified size (including
+   * stateful old and older properties.
+   *
+   * Warning:
+   * If size is greater than the current size, the new properties are set
+   * to NULL except * for the last element - which is set to a blank material
```

Comment:
The issue is with the ``MaterialData::resizeProps`` function.  ``_props`` member and friends are basically vectors of ``PropertyValue*``  and the resize function calls resize on the vector, (inserting NULLs to any new elements), and then proceeds to reset/fix the last element only as a blank material property value using new.